lib/portlayer/exec/container.go

```diff
@@ -451,8 +452,21 @@ func (c *Container) Remove(ctx context.Context, sess *session.Session) error {
 		return c.vm.DeleteExceptDisks(ctx)
 	})
 	if err != nil {
-		c.State = existingState
-		return err
+		f, ok := err.(types.HasFault)
+		if !ok {
+			c.State = existingState
+			return err
+		}
+		switch f.Fault().(type) {
+		case *types.InvalidState:
+			log.Warnf("container VM is in invalid state, unregistering")
+			if err := c.vm.Unregister(ctx); err != nil {
+				log.Errorf("Error while attempting to unregister container VM: %s", err)
+				return err
+			}
+		default:
+			log.Debugf("Fault while attempting to destroy vm: %#v", f.Fault())
```

Comment:
Not sure what Faults could apply here - should we do the follow as in the non fault error case?:
```
c.State = existingState
return err
```